src/random.cpp

```diff
@@ -84,33 +90,53 @@ uint64_t Random::next(uint64_t max) {
 #endif
 
   uint64_t get_random_seed(uint64_t seed) {
+#if defined(HAVE_ARC4RANDOM)
+    arc4random_buf(&seed, sizeof(seed));
+#else
     static const char* device = "/dev/urandom";
-
-    int fd = open(device, O_RDONLY);
-
-    if (fd < 0) {
+    ssize_t num_bytes;
+    bool readurandom = true;
+#if defined(HAVE_GETRANDOM)
+    num_bytes = static_cast<ssize_t>(syscall(SYS_getrandom, &seed,
+            sizeof(seed), GRND_NONBLOCK));
+    if (num_bytes < static_cast<ssize_t>(sizeof(seed))) {
       char buf[STRERROR_BUFSIZE_];
       char* err = STRERROR_R_(errno, buf, sizeof(buf));
-      LOG_CRITICAL("Unable to open random device (%s): %s", device, err);
-      return seed;
+      LOG_WARN("Unable to read %zu random bytes (%s): %zu read",
+            sizeof(seed), err, num_bytes);
+    } else {
+      readurandom = false;      
     }
+#endif
```

Comment:
It would improve readability if this was `#endif // defined(HAVE_GETRANDOM)`. These nested `#if`s are hard to follow.